addons/crm/models/crm_lead.py

```diff
@@ -1802,27 +1802,13 @@ def opps_key(opportunity):
     # CUSTOMER TOOLS
     # --------------------------------------------------
 
-    def _find_matching_partner(self, email_only=False):
-        """ Try to find a matching partner with available information on the
-        lead, using notably customer's name, email, ...
-
-        :param email_only: Only find a matching based on the email. To use
-            for automatic process where ilike based on name can be too dangerous
-        :return: partner browse record
-        """
+    def _find_matching_partner(self):
         self.ensure_one()
         partner = self.partner_id
-
-        if not partner and self.email_from:
-            partner = self.env['res.partner'].search([('email', '=', self.email_from)], limit=1)
-
-        if not partner and not email_only:
-            # search through the existing partners based on the lead's partner or contact name
-            # to be aligned with _create_customer, search on lead's name as last possibility
-            for customer_potential_name in [self[field_name] for field_name in ['partner_name', 'contact_name', 'name'] if self[field_name]]:
-                partner = self.env['res.partner'].search([('name', 'ilike', customer_potential_name)], limit=1)
-                if partner:
-                    break
+        normalize_email = tools.email_normalize(self.email_from)
+        if not partner and normalize_email:
+            # some of the tests contain formatted emails, and it appears possible to have formatted emails in partner emails so used ilike
+            partner = self.env['res.partner'].search([('email', 'ilike', normalize_email)], limit=1)
```

Comment:
Note that an improvement if we have multiple partners (to avoid the limit=1) would be to filter on company, if set, for example.

Reply:
hello @tde-banana-odoo :wave: ,
There may be duplicate contacts with the same company and email. 
What do you think about it?

BTW tried something like this:
```.py
domain = [('email', 'ilike', self.email_normalized)]
if self.company_id:
    domain.append(self.env['crm.lead']._check_company_domain(self.company_id))
partner = self.env['res.partner'].search(domain) 
```

Thanks

Reply:
I would sort more than trying to compute a domain, like sort by matching company first, then no company, then other company, something like that :) . We don't really need a complex domain, sorting / filtering should be sufficient.